services/licenseStore.js

```diff
@@ -26,24 +26,56 @@ function buildPortedLicense(license, string, url) {
 }
 
 function buildLicensesIndex(licenses) {
-  return licenses.reduce((idx, license) => Object.assign(idx, { [license.id]: license }), {});
+  return licenses.reduce((idx, license) => {
+    Object.assign(idx, { [license.id]: license });
+    return idx;
+  }, {});
+}
+
+function buildLicenseNamesIndex(licenses) {
+  return licenses.reduce((idx, license) => {
+    const page = idx[license.name];
+    if (page) {
+      page.push(license.id);
+    } else {
+      Object.assign(idx, { [license.name]: [license.id] });
+    }
+    return idx;
```

Comment:
```
nameToLicenseMapping = {};
licenses.forEach(license => {
  licensesWithSameName = nameToLicenseMapping[license.name]
  if (licensesWithSameName)
    licensesWithSameName.push(license)
  } else {
    nameToLicenseMapping[license.name] = [license]
  }
});
```

we just talked about this snipped. probably reduce (as in your code) is cool, but we could borrow naming and the non-usage ob Object.assign from my snippet.
```